### ISSUE TITLE
Decide a formatting style guide for the project

### ISSUE THREAD

**Issue by VaiTon** [openfoodfacts/openfoodfacts-server#2044]

**Comment by VaiTon** [comment]
I would like to hear your opinion @openfoodfacts/openfoodfacts-server

**Comment by CharlesNepote** [comment]
We use .editor config file. Isn't it ok? Did you read: https://en.wiki.openfoodfacts.org/Software_Development ?

**Comment by VaiTon** [comment]
Yeah .editorconfig is ok. The problem is that for perl it doesn't enforce any coding style and so any consistency. In a lot of files you will see formatting inconsistencies like these.

Here the method is called without leaving space inside the parenthesis.
https://github.com/openfoodfacts/openfoodfacts-server/blob/855a82868da529290f7947358b0da70a801c91fe/t/food_normalize_serving_size.t#L24

Here we left a space right before the parenthesis. Also the method call is closed on next line.
https://github.com/openfoodfacts/openfoodfacts-server/blob/855a82868da529290f7947358b0da70a801c91fe/t/additives_tags.t#L37-L38

**Comment by hangy** [comment]
I think it could be a good idea to decide on a code style and to enforce it with [`Test::PerlTidy`](https://metacpan.org/pod/Test::PerlTidy) during CI.

Unsure about the "best" style to choose. Have you tested which options cause the fewest changes to the current source code? It might also be good to not report code style variations as errors right away, because re-formatting a complete codebase in one go isn't a good idea and will break most open PRs.

**Comment by VaiTon** [comment]
Maybe the GNU style would cause the fewest changes but it indeed they are a lot.

**Comment by CharlesNepote** [comment]
Is there a way to "convert" files only if they are modified?

Would the first step be to validate all PRs?

**Comment by hangy** [comment]
> Would the first step be to validate all PRs?

I like the idea! With GitHub Actions this could be possible, as demonstrated by https://github.com/lots0logs/gh-action-get-changed-files/blob/master/entrypoint.js
One only potential problem could be the size of ProductOpener modules. If we changed a line in Display.pm, the other thousands of lines of code would also be run by perltidy.
